src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java

```diff
@@ -146,7 +153,37 @@ public MediaObject getQoSObject() {
 
     @Override
     public Double getCurrentPlaybackTime() {
-      return null;
+      if (initialTime == 0) {
+        setInitialTime();
+      }
+      if (!isPaused) {
+        incrementPlayheadPosition();
+        return (double) playheadPosition;
+      }
+      return (double) pausedPlayheadPosition;
+    }
+
+    private void setInitialTime() {
+      this.initialTime = System.currentTimeMillis();
+    }
+
+    private void incrementPlayheadPosition() {
+      this.playheadPosition = ((System.currentTimeMillis() - initialTime) / 1000) - offset;
```

Comment:
I think this should have a comment describing what the logic here is.

Reply:
Something like this https://github.com/segmentio/api/blob/master/packages/api-transforms/lib/transform.js#L474-L502